<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>730</width>
    <height>830</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>730</width>
    <height>830</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>730</width>
    <height>830</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
   <widget class="QFrame" name="frame_uart">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>40</y>
      <width>221</width>
      <height>261</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>100</width>
      <height>100</height>
     </size>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_comid">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>PortName</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_box_com">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>60</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_baud">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>BaudRate</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_box_baud">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>90</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_data">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>DataBits</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_box_data">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>120</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_stop">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>StopBits</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_box_stop">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>150</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_parity">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Parity</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_box_parity">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>180</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_uart_open">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>220</y>
       <width>85</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>open</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_uart_close">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>220</y>
       <width>85</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>close</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_dev_id_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Serial Control</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_dev">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>370</y>
      <width>221</width>
      <height>81</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_dev_id">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>ID</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="line_edit_dev_id">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_dev_id_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Helmet ID</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_socket">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>530</y>
      <width>221</width>
      <height>211</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_server_ipaddr">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Server IP</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="line_edit_ipaddr">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>192.168.1.251</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_server_port">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>TCP port</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tcp_port">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>110</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>8000</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_socket_open">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>85</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>connect</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_socket_close">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>170</y>
       <width>85</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>disconnect</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_local_ipaddr">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Local IP</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="line_edit_local_ipaddr">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>192.168.1.105</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_dev_id_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Socket Control</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_server_port_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>UDP port</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="udp_port">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>140</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>8001</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>481</width>
      <height>771</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="deviceTab">
     <property name="autoFillBackground">
      <bool>true</bool>
     </property>
     <attribute name="title">
      <string>Device Management </string>
     </attribute>
     <widget class="QFrame" name="frame_info">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>481</width>
        <height>431</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_led">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>LED Control</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_led_on">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>10</y>
         <width>90</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>open</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_led_off">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>10</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>close</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_beep">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>BEEP Control</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_beep_on">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>70</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>open</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_beep_off">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>70</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>close</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="text_edit_recv">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>170</y>
         <width>481</width>
         <height>261</height>
        </rect>
       </property>
       <property name="acceptDrops">
        <bool>false</bool>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_stauts">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Internal status</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_refresh">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>130</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>refresh</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_reboot">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>71</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_reboot">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>100</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>reset</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="LEDBox_red">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>92</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>red</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="LEDBox_blue">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>40</y>
         <width>92</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>blue</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="LEDBox_green">
       <property name="geometry">
        <rect>
         <x>230</x>
         <y>40</y>
         <width>92</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>green</string>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_test">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>430</y>
        <width>481</width>
        <height>331</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_cmd">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>131</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Custom instruction</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="line_edit_cmd">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>311</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_send_cmd">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>30</y>
         <width>141</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Command sending</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_test">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>111</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Debug window</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="text_edit_test">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>471</width>
         <height>171</height>
        </rect>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="btn_clear">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>280</y>
         <width>92</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>Clear data</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>thermo-hygrometer</string>
     </attribute>
     <widget class="ThermWnd" name="therm_widget" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>100</y>
        <width>231</width>
        <height>631</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="therm_lcd">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>40</y>
        <width>161</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="hygro_lcd">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>40</y>
        <width>161</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="HygroWnd" name="hygro_widget" native="true">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>100</y>
        <width>211</width>
        <height>631</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>0</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>thermometer</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>0</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>hygrometer</string>
      </property>
     </widget>
    </widget>
    <widget class="GPSInfoWnd" name="gps_wnd">
     <attribute name="title">
      <string>GPS</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>730</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ThermWnd</class>
   <extends>QWidget</extends>
   <header location="global">thermwnd.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HygroWnd</class>
   <extends>QWidget</extends>
   <header location="global">hygrownd.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GPSInfoWnd</class>
   <extends>QWidget</extends>
   <header location="global">gpsinfownd.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
